src/misc/cross.cpp

```diff
@@ -715,6 +715,88 @@ std::string cfstr_to_string(CFStringRef source)
 	return lang;
 }
 
+#if defined (__linux__)
+
+#include <fcntl.h>
+
+int get_num_physical_cpus()
+{
+	constexpr int MaxCpus = 128;
+	bool core_ids[MaxCpus] = {};
```

Comment:
How about using `std::set<int> all_core_ids` instead of `core_ids[MaxCpus]` and `int num_cores`? Then:

```c++
...
if (core_id >= 0) {
    all_core_ids.insert(core_id);
}
...
```
and at the end `all_core_ids.size()` contains the number of cores.

Reply:
The array is a lot more efficient, not that matters too much for this probing code that runs once or twice.  You can think of it as a "perfect hash map" in that the values are already normalized to a reasonable range and you don't actually have to hash anything.  Also `std::unordered_set` is supposed to be more performant than `std::set` unless you really need the ordering for some reason.

Reply:
Yeah, I prefer the array. Using a set is IMO overdoing it.